### PR TITLE
[feat] Add health checks

### LINKED ISSUE #366: Provide a way to integrate with AspNetCore.Diagnostics.HealthChecks
**Is your feature request related to a problem? Please describe.**
The latest version (7.1.0) of `AspNetCore.Diagnostics.HealthChecks` for CosmosDB requires that the CosmosClient be made available to the HealthCheck Api.

See breaking changes here.
https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/src/HealthChecks.CosmosDb/README.md#breaking-changes

**Describe the solution you'd like**
Consider a way to either integrate the health check package, or a way to expose the CosmosClient to allow it be passed to the HealthCheck API

**Describe alternatives you've considered**
Could maybe expose public property on `ICosmosClientProvider ` to expose the CosmosClient, and make `ICosmosClientProvider ` public?

### PR DESCRIPTION
**Pull request by philip-reed**
Adds functionality discussed in #366.

Closes #366 

Adds support for [AspNet Core Health Checks](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks) using the [HealthChecks.CosmosDb](https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/src/HealthChecks.CosmosDb/README.md) package.

## Setup

To configure Cosmos DB health checks:  

```csharp
services.AddHealthChecks().AddCosmosRepository();
```

By default, this will scan all of the assemblies in your solution to locate the container names for each of your `IItem` types.  To refine this, and potentially reduce startup times, pass in the Assemblies containing your `IItem` types:

```csharp
services.AddHealthChecks().AddCosmosRepository(assemblies: typeof(ExampleItem).Assembly);
```

The Cosmos Repository Health package supports all of the existing functionality of Health Checks, such as failureStatus and tags, see the [Microsoft Documentation](https://learn.microsoft.com/en-us/aspnet/core/host-and-deploy/health-checks) for configuration details.

### CODE REVIEW

**Comment by mumby0168** [comment]
@all-contributors please add @philip-reed for docs, code and tests

File: src/Microsoft.Azure.CosmosRepository.AspNetCore/Microsoft.Azure.CosmosRepository.AspNetCore.csproj [hunk 1]
```diff
@@ -37,31 +36,30 @@
...
     <ItemGroup>
-        <PackageReference Include="Microsoft.Extensions.Hosting.Abstractions" />
         <PackageReference Include="MinVer">
             <PrivateAssets>all</PrivateAssets>
             <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
         </PackageReference>
         <PackageReference Include="Scrutor" />
+        <PackageReference Include="AspNetCore.HealthChecks.CosmosDb"  />
+    </ItemGroup>
+    <ItemGroup>
```
Comment (mumby0168) [inline src/Microsoft.Azure.CosmosRepository.AspNetCore/Microsoft.Azure.CosmosRepository.AspNetCore.csproj:49]: @IEvangelist spoke with @philip-reed around this change, and I am not sure when I implemented this the first time why I didn't just do a framework reference as this was the idea of the .AspNetCore package. I think this is good and not a problem, however, I just wonder if it's considered a breaking change or not?

**Review by mumby0168 — APPROVED** [review]
LGTM, nice work.

File: src/Microsoft.Azure.CosmosRepository.AspNetCore/Extensions/HealthChecksBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    {
+        builder.AddAzureCosmosDB(
+            clientFactory: provider => provider.GetRequiredService<ICosmosClientProvider>().CosmosClient,
+            optionsFactory: sp =>
+            {
+                IOptions<RepositoryOptions> options = sp.GetRequiredService<IOptions<RepositoryOptions>>();
+                ICosmosItemConfigurationProvider itemConfigProvider =
+                    sp.GetRequiredService<ICosmosItemConfigurationProvider>();
+                IEnumerable<string> containers = itemConfigProvider.GetAllItemConfigurations(assemblies)
+                    .Select(i => i.ContainerName).Distinct();
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosRepository.AspNetCore/Extensions/HealthChecksBuilderExtensions.cs:47]:
```suggestion
                    .DistinctBy(i => i.ContainerName);
```

File: src/Microsoft.Azure.CosmosRepository.AspNetCore/Microsoft.Azure.CosmosRepository.AspNetCore.csproj (hunk 1 above)
Comment (IEvangelist) [reply src/Microsoft.Azure.CosmosRepository.AspNetCore/Microsoft.Azure.CosmosRepository.AspNetCore.csproj:49]: I don't believe it's a breaking change, it just adds a new dependency - but I could be wrong. I'm good with this...

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**